pkg/controller/atlascustomrole/custom_role.go

```diff
@@ -0,0 +1,119 @@
+package atlascustomrole
+
+import (
+	"fmt"
+	"reflect"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/customroles"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+type roleController struct {
+	ctx       *workflow.Context
+	projectID string
+	service   customroles.CustomRoleService
+	role      *akov2.AtlasCustomRole
+	dpEnabled bool
+	k8sClient client.Client
+}
```

Comment:
This, more than a controller, is a reconciliation request object, becase it is only tied to a single request context.

Calling it a controller makes it also confusing is this a sub-controller/sub-reconciler or the custom roles reconciler?